saleor/graphql/order/mutations/orders.py

```diff
@@ -541,3 +559,183 @@ def perform_mutation(cls, root, info, **data):
             order_captured(order, info.context.user, payment.total, payment)
         order_confirmed(order, info.context.user, send_confirmation_email=True)
         return OrderConfirm(order=order)
+
+
+class OrderLinesCreate(BaseMutation):
+    order = graphene.Field(Order, description="Related order.")
+    order_lines = graphene.List(
+        graphene.NonNull(OrderLine), description="List of added order lines."
+    )
+
+    class Arguments:
+        id = graphene.ID(
+            required=True, description="ID of the order to add the lines to."
+        )
+        input = graphene.List(
+            OrderLineCreateInput,
+            required=True,
+            description="Fields required to add order lines.",
+        )
+
+    class Meta:
+        description = "Create order lines for an order."
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = OrderError
+        error_type_field = "order_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        order = cls.get_node_or_error(info, data.get("id"), only_type=Order)
+        if order.status not in ORDER_EDITABLE_STATUS:
+            raise ValidationError(
+                {
+                    "id": ValidationError(
+                        "Only draft and unconfirmed orders can be edited.",
+                        code=OrderErrorCode.NOT_EDITABLE,
+                    )
+                }
+            )
+
+        lines_to_add = []
+        for input_line in data.get("input"):
+            variant_id = input_line["variant_id"]
+            variant = cls.get_node_or_error(
+                info, variant_id, "variant_id", only_type=ProductVariant
+            )
+            quantity = input_line["quantity"]
+            if quantity > 0:
+                if variant:
+                    lines_to_add.append((quantity, variant))
+            else:
+                raise ValidationError(
```

Comment:
In this place, we should build a list of invalid `variant_ids` and raise error outside for loop.